sql/analyze_test.go

```diff
@@ -508,7 +545,7 @@ func TestSimplifyOrExprCheck(t *testing.T) {
 		{`a = 1 OR a = 1`, `a = 1`},
 		{`a = 1 OR a = 2`, `a IN (1, 2)`},
 		{`a = 2 OR a = 1`, `a IN (1, 2)`},
-		{`a = 1 OR a != 1`, `true`},
+		{`a = 1 OR a != 1`, `a IS NOT NULL`},
```

Comment:
Simplifying `a = 1 OR a != 1` to `true` was a bug. Those expressions are not equivalent if `a` is `NULL`. This was caught be the enhancements to `checkEquivExpr` to test the `NULL` value in the expressions. Yay tests!